dev/InfoBar/InfoBar.xaml

```diff
@@ -0,0 +1,201 @@
+                                        <Setter Target="CloseButton.Visibility" Value="Collapsed"/>
+                                    </VisualState.Setters>
+                                </VisualState>
+                            </VisualStateGroup>
+
+                            <VisualStateGroup x:Name="InfoBarVisibility">
+                                <VisualState x:Name="InfoBarVisible"/>
+                                <VisualState x:Name="InfoBarCollapsed">
+                                    <VisualState.Setters>
+                                        <Setter Target="ContentRoot.Visibility" Value="Collapsed"/>
+                                    </VisualState.Setters>
+                                </VisualState>
+                            </VisualStateGroup>
+                            
+                        </VisualStateManager.VisualStateGroups>
+
+                        <Grid
+                            HorizontalAlignment="Stretch"
+                            MinHeight="{ThemeResource InfoBarMinHeight}"
+                            Background="{TemplateBinding Background}"
+                            Padding="{StaticResource InfoBarContentRootPadding}">
+
+                            <Grid.ColumnDefinitions>
+                                <ColumnDefinition Width="Auto"/> <!-- Icon -->
+                                <ColumnDefinition Width="*"/>    <!-- Title, message, and action -->
+                                <ColumnDefinition Width="Auto"/> <!-- Close button -->
+                            </Grid.ColumnDefinitions>
+
+                            <Grid.RowDefinitions>
+                                <RowDefinition Height="Auto"/>
+                                <RowDefinition Height="Auto"/>
+                            </Grid.RowDefinitions>
+
+                            <TextBlock x:Name="StandardIcon"
+                                Grid.Column="0"
+                                Visibility="Collapsed"
+                                VerticalAlignment="Top"
+                                Margin="{StaticResource InfoBarIconMargin}"
+                                FontSize="{StaticResource InfoBarIconFontSize}"
+                                Text="{StaticResource InfoBarInformationalIconGlyph}"
+                                FontFamily="{ThemeResource SymbolThemeFontFamily}"
+                                AutomationProperties.AccessibilityView="Raw" />
+
+                            <Border x:Name="UserIconBorder"
+                                Grid.Column="0"
+                                Visibility="Collapsed"
+                                VerticalAlignment="Top"
+                                Margin="{StaticResource InfoBarIconMargin}"
+                                Child="{Binding RelativeSource={RelativeSource TemplatedParent}, Path=TemplateSettings.IconElement}" />
+
+                            <primitives:InfoBarPanel
```

Comment:
Could we use VariableSizedWrapGrid instead of a custom panel? If we only want to put the message/content below the title if there is not enough space, VariableSizedWrapGrid would also allow us to do that without having to introduce a new primitive here.

Reply:
Well, the problem is the action button as well. The design says that if they don't all fit horizontally, then layout needs to be vertical. So if you had a long title but a short message + the button, the message and button wouldn't be on the same line.

Reply:
Hmm I see, guess a new control is easier then.

Reply:
Maybe... you've only seen the bugs currently in it, not the many many strange things it's done up to this point...

Reply:
Haha, well let's stick with a new primitive here then, that way we can fix bugs easier (at least until WinUI 3 ...).

Reply:
Spacing is also an issue, there's different spacing depending on which way layout flows and that's an unusual thing to handle.

Reply:
Yeah, I noticed the spacing things. How would that play with users setting margin on their custom content? That wouldn't cause many issues I assume?

Reply:
The custom content area shouldn't be affected, I think. Just the title/message/action button.